pkg/auth/rbac.go

```diff
@@ -38,14 +49,114 @@ type Rbac struct {
 	policyStore *policyStore
 }
 
+type Identity struct {
+	trustDomain    string
+	namespace      string
+	serviceAccount string
+}
+
+type rbacConnection struct {
+	srcIdentity Identity
+	srcNetwork  string
+	srcIp       []byte
+	dstIp       []byte
+	dstPort     uint32
+}
+
 func NewRbac() *Rbac {
 	return &Rbac{
 		policyStore: newPolicystore(),
 	}
 }
 
-func (r *Rbac) DoRbac(conn *RbacConnection) bool {
-	workload := conn.fetchWorkload()
+func (r *Rbac) Run(ctx context.Context) {
+	reader, err := ringbuf.NewReader(bpf.ObjWorkload.KmeshWorkload.SockOps.MapOfTuple)
+	if err != nil {
+		log.Errorf("open ringbuf map FAILED, err: %v", err)
+		return
+	}
+	defer func() {
+		if err := reader.Close(); err != nil {
+			log.Errorf("reader Close FAILED, err: %v", err)
+		}
+	}()
+
+	rec := ringbuf.Record{}
+	tupleV4, tupleV6 := bpfSockTupleV4{}, bpfSockTupleV6{}
+	var conn rbacConnection
+	for {
+		select {
+		case <-ctx.Done():
+			return
+		default:
+			if err = reader.ReadInto(&rec); err != nil {
+				log.Errorf("ringbuf reader FAILED to read, err: %v", err)
+				continue
+			}
+
+			if len(rec.RawSample) != MSG_LEN {
+				log.Errorf("wrong length %v of a msg...", len(rec.RawSample))
+				continue
+			}
+			msgType := binary.LittleEndian.Uint32(rec.RawSample)
+			var buf *bytes.Buffer
+			switch msgType {
+			case MSG_TYPE_IPV4:
+				buf = bytes.NewBuffer(rec.RawSample[4 : IPV4_TUPLE_LENGTH+4+1])
+				if err = binary.Read(buf, binary.LittleEndian, &tupleV4); err != nil {
+					log.Errorf("deserialize IPv4 FAILED, err: %v", err)
+					continue
+				}
+				conn = buildConnV4(&tupleV4)
+			case MSG_TYPE_IPV6:
+				buf = bytes.NewBuffer(rec.RawSample[4:])
+				if err = binary.Read(buf, binary.LittleEndian, &tupleV6); err != nil {
+					log.Errorf("deserialize IPv6 FAILED, err: %v", err)
+					continue
+				}
+				conn = buildConnV6(&tupleV6)
+			default:
+				log.Errorf("INVALID msg type: %v", msgType)
+				continue
+			}
+
+			if !r.doRbac(&conn) {
+				switch msgType {
+				case MSG_TYPE_IPV4:
+					if err = xdpHandlerUpdateV4(&xdpHandlerKeyV4{Tuple: tupleV4}); err != nil {
```

Comment:
This function name should be more clear It is to notify xdp to rst. So may call `notifyXdpConRst`/`notifyConRst`, etc

Reply:
done